src/content/blog/scripts-powered-by-ddev.md

```diff
@@ -0,0 +1,78 @@
+
+Some, however, attempted DDEV but found themselves disoriented by its distinct operation, configuration options, and the overall idea of a project residing within a container.
+
+## Scripts powered by DDEV
+
+So, I thought that in order to make DDEV more user friendly with them, I had to wrap DDEV in a script program that would really adapt to the development team's needs. A common set
+of operations that were done every day were things such: installing a shop (with different versions), installing the plugin, configuring the plugin, configuring Ngrok,
+clear caches, backup & restore the database...
+It would be great to have a tool that could do all these operations automatically... for all the different shops that we support. It would require a lot of work, but
+thanks to the command line capabilities that DDEV offer, it would be totally possible to do. Because DDEV is fully managed using the command line, and it offers a lot options.
+Too many options, perhaps, for a newbie. But all these command line options give the flexibility that is needed for building a strong script program that runs powered by DDEV.
+
+I started, then, working in such a tool, which I visualized as a local shop manager (lsm) with the following set of goals:
+
+- Each needed operation would translate in a command for lsm, like: initialize the containers, install the shop in the containers, remove them, download a specific version of the plugin, install the plugin, uninstall the plugin, configure the plugin, etc...
+- Strong focus in flexibility, with a lot of settings supported and that each command can declare. All the settings will be provided in a yaml file, or by CLI arguments.
+- Provide as many default settings values as possible to ensure that lsm works in most of the cases out of the box.
+- Total abstraction from the shop systems: the workflow for managing a shop (magento, woocomerce, prestashop, etc...) would be exactly the same regardless which shop is being managed. So the end users need to learn one workflow of lsm commands and they can then manage all the shops.
+- Make the tool as easy to use as possible, with an independent documentation. Starting with a 'help' command that can describe all the other commands and the options that are accepted.
+- Built with latests PHP features, using Symfony project when needed (yaml...), PHPUnit...
+- Use Phar format for distributing the lsm tool, which also has its own version lifecycle and self-update options.
+- Possibility of having several instances installed at the same time without any conflict.
+
+Under the hood, every command is retrieving the settings from the environment, doing necessary changes using PHP (change folders, create files...), and then calling the appropriate ddev command for the real action: config, start, stop, export-db, import-db, exec, share, composer... The tool abstracts DDEV for most of the users, because they do not need to use the ddev CLI tool for most of the operations...
```

Comment:
```suggestion
Under the hood, each command retrieves the settings from the environment, doing necessary changes using PHP (change folders, create files...), and then calling the appropriate DDEV command for the real action: `config`, `start`, `stop`, `export-db`, `import-db`, `exec`, `share`, `composer`... The tool abstracts DDEV for most of the users, because they do not need to use the ddev CLI tool for most of the operations.
```